File: 3550-smallest-index-with-digit-sum-equal-to-index/3550-smallest-index-with-digit-sum-equal-to-index.c
int digitCount(int num){
    int sum = 0;
    while(num > 0){
        int digit = num % 10;
        sum += digit;
        num = num / 10;
    }
    return sum;
}

int smallestIndex(int* nums, int numsSize) {
    for(int i = 0 ; i< numsSize ; i++){
        if( i == digitCount(nums[i])){
            return i;
        }
    }
    return -1;   
}

File: 3550-smallest-index-with-digit-sum-equal-to-index/3550-smallest-index-with-digit-sum-equal-to-index.py
class Solution:
    def digitCount(self,num):
        total = 0
        while(num > 0):
            digit = num % 10
            total += digit
            num = num // 10
        return total

    def smallestIndex(self, nums: List[int]) -> int:
        for i in range(len(nums)):
            if( i == self.digitCount(nums[i])):
                return i
        return -1